core/src/test/java/org/testcontainers/utility/TestcontainersConfigurationTest.java

```diff
@@ -156,13 +155,6 @@ public void shouldReadDockerClientStrategyFromEnvironment() {
         assertEquals("Docker client strategy is changed by env var", "foo", newConfig().getDockerClientStrategyClassName());
     }
 
-    @Test
-    public void shouldUseImplicitDockerClientStrategyWhenDockerHostPropertyIsSet() {
-        userProperties.remove("docker.client.strategy");
-        userProperties.put("docker.host", "tcp://1.2.3.4:5678");
-        assertEquals("Docker client strategy is implicitly set when docker host property is set", EnvironmentAndSystemPropertyClientProviderStrategy.class.getCanonicalName(), newConfig().getDockerClientStrategyClassName());
-    }
-
```

Comment:
Aren't we introducing a regression here?

Reply:
No.
The problem is, this test is an isolated unit test, looking at `TestcontainerConfiguration` class.

However, this is actually handled in `EnvironmentAndSystemPropertyClientProviderStrategy` and should not introduce a regression. This is correspondingly tested here now:
https://github.com/testcontainers/testcontainers-java/pull/4387/files#diff-4eda76cf7611dd2d1507bb655e7486223e364b121a335bedab32264602b52054R99

Reply:
Yeah IIRC this test was doubly-covering the behaviour that existed before. Maybe we could reinstate the test and check that it passes though?

Reply:
But it will not pass like this, since we ripped out the code from `TestcontainersConfiguration`:
https://github.com/testcontainers/testcontainers-java/pull/4387/files#diff-87c8b4b1adf608eaf354a256ad4a5af86049d324cc351c6f6f6b94e96232cbd7L185

However, also some unit tests in `EnvironmentAndSystemPropertyClientProviderStrategyTest` are failing, I need a bit to get my head back into the context, sorry.

Reply:
Ok found the reason and basically have the working stubbing. Removing this test is IMO no regression since we have indeed the corresponding new test in `EnvironmentAndSystemPropertyClientProviderStrategyTest`.

Since I needed to stub/fake `DefaultDockerClientConfig.Builder` I added an additional package-private constructor for testing, is this approach ok?